library/alloc/src/vec/mod.rs

```diff
@@ -95,59 +96,82 @@ mod drain;
 #[cfg(not(no_global_oom_handling))]
 mod cow;
 
-#[cfg(not(no_global_oom_handling))]
 pub(crate) use self::into_iter::AsIntoIter;
 #[stable(feature = "rust1", since = "1.0.0")]
 pub use self::into_iter::IntoIter;
 
 mod into_iter;
 
-#[cfg(not(no_global_oom_handling))]
 use self::is_zero::IsZero;
 
 mod is_zero;
 
-#[cfg(not(no_global_oom_handling))]
 mod source_iter_marker;
 
 mod partial_eq;
 
-#[cfg(not(no_global_oom_handling))]
 use self::spec_from_elem::SpecFromElem;
 
-#[cfg(not(no_global_oom_handling))]
 mod spec_from_elem;
 
-#[cfg(not(no_global_oom_handling))]
 use self::set_len_on_drop::SetLenOnDrop;
 
-#[cfg(not(no_global_oom_handling))]
 mod set_len_on_drop;
 
-#[cfg(not(no_global_oom_handling))]
 use self::in_place_drop::InPlaceDrop;
 
-#[cfg(not(no_global_oom_handling))]
 mod in_place_drop;
 
-#[cfg(not(no_global_oom_handling))]
 use self::spec_from_iter_nested::SpecFromIterNested;
 
-#[cfg(not(no_global_oom_handling))]
 mod spec_from_iter_nested;
 
-#[cfg(not(no_global_oom_handling))]
 use self::spec_from_iter::SpecFromIter;
 
-#[cfg(not(no_global_oom_handling))]
 mod spec_from_iter;
 
-#[cfg(not(no_global_oom_handling))]
 use self::spec_extend::SpecExtend;
 
-#[cfg(not(no_global_oom_handling))]
 mod spec_extend;
 
+pub(crate) trait Operations {
+    type Error;
+    fn capacity_overflow() -> Self::Error;
+    fn alloc_error(layout: Layout) -> Self::Error;
+}
```

Comment:
Last idea that occurred to me:
```suggestion
pub(crate) trait VecError {
    fn capacity_overflow() -> Self;
    fn alloc_error(layout: Layout) -> Self;
}
```
`Self` is never actually used before, so just making the class directly about errors should make type inference work better.

Reply:
Yep, that also cleaned up a lot of code :)